### PR TITLE
feat: user-installable apps

### PR DESCRIPTION
**Pull request by AtoraSuunva**
**Please describe the changes this PR makes and why it should be merged:**

Adds in the new fields & deprecation changes from https://discord.com/developers/docs/change-log#userinstallable-apps-preview

Now you can make user-installable apps with type-safety :)

Marked as draft until these issues are resolved, since what the API returns and what the docs say is different (current PR matches the API, will update based on issues):
- https://github.com/discord/discord-api-docs/issues/6744
	- https://github.com/discord/discord-api-docs/pull/6765
- https://github.com/discord/discord-api-docs/issues/6730

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
- https://github.com/discord/discord-api-docs/pull/6723
- https://github.com/discord/discord-api-docs/pull/6782

### CODE REVIEW

**Comment by kyranet** [comment]
The last issue just got resolved. 👀

**Comment by AtoraSuunva** [comment]
Types now match the API and docs.

I'm unsure if this should be merged (as the docs say: "API details are subject to change.") but I'll leave it up to the maintainers to decide.

If not yet, I'll keep this updated with changes.

**Comment by kyranet** [comment]
You can mark it as `@unstable` in that case.

**Comment by vladfrangu** [comment]
@AtoraSuunva boop, do you have some time, or would you like some help with this?

**Comment by AtoraSuunva** [comment]
@vladfrangu oops, got busy with some things. Used https://github.com/discord/discord-api-docs/pull/6782 as reference for what to mark as `@unstable`

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Is this an API v10 only feature? If yes, please let us know! If not, ctrl c ctrl v in api v9 too please

File: deno/payloads/v10/_interactions/applicationCommands.ts
Comment (Syjalo) [inline deno/payloads/v10/_interactions/applicationCommands.ts:80]: I don't think this shoud be deprecated rn. Because contexts property is unstable.
```suggestion
	// TODO: deprecate after APIApplicationCommand.contexts become stable
	/**
	 * Indicates whether the command is available in DMs with the app, only for globally-scoped commands. By default, commands are visible
	 */
	dm_permission?: boolean;
```

**Review by Syjalo — CHANGES_REQUESTED** [review]

File: deno/payloads/v10/_interactions/applicationCommands.ts
Comment (Jiralite) [reply deno/payloads/v10/_interactions/applicationCommands.ts:80]: If this is to reflect Discord pedantically, perhaps it should be deprecated.

Deprecating a property for a field in beta seems strange too, though.

**Comment by AtoraSuunva** [comment]
> Is this an API v10 only feature? If yes, please let us know! If not, ctrl c ctrl v in api v9 too please

Seems to work even all the way back to v6, both REST requests and gateway. Only added it to v9 for now though.

File: deno/payloads/v10/_interactions/applicationCommands.ts
Comment (AtoraSuunva) [reply deno/payloads/v10/_interactions/applicationCommands.ts:80]: It's odd but I did it to reflect exactly what the API docs say https://discord.com/developers/docs/interactions/application-commands#application-command-object

Depends how strictly we want to stay with the docs.

Comment (vladfrangu) [reply deno/payloads/v10/_interactions/applicationCommands.ts:80]: I don't wanna block this PR any longer, lets stick to what the docs say. If they undeprecate dm_permission, so will we. Its not the end of the world :D

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**
